ALGORITMO:
1) Função de entrada e verificação dos dados:
   Essa função deve criar e organizar duas listas para armazenar os dados das (n) equações do sistema  escolhido pelo usuário. Esse sistema tem n(n+1)/2 coeficientes, n incógnitas e n termos independentes. 
   
                Obtenção do Número de Equações:

Solicita ao usuário a quantidade (n) de equações no sistema.

                Validação do Número de Equações:

Verifica se o número de equações é um inteiro positivo. Caso contrário, solicita correção.


        Obtenção e Validação dos Coeficientes:

Utiliza dois loops para obter os (n(n+1)/2) coeficientes de cada equação.
Garante que os coeficientes sejam números reais. Caso contrário, solicita correção.
Verifica se a matriz resultante é triangular inferior.

            Obtenção e Validação dos Termos Independentes:

Utiliza um loop para obter os n termos independentes de cada equação.
Garante que os termos independentes sejam números reais. Caso contrário, solicita correção.

                    Confirmação dos Valores:

Permite ao usuário confirmar os valores inseridos, digitando 1 para confirmar e 0 para retornar.

                       Retorno das Listas A e B:

Se a confirmação for 1, retorna as listas de coeficientes (A) e termos independentes (B).
Se a confirmação for 0, reinicia o processo.

2) Função de resolução do sistema triangular:
   Essa função deve receber as listas A e B geradas pela última função e transformá-las em uma nova lista X, que contêm os resultados das (n) incognitas em ordem. 
            
                Obtenção do Número de Equações:

Obtém o número de equações (n) a partir da lista B.

            Inicialização da Lista de Soluções (X):

Inicializa uma lista X preenchida com zeros.
   
        Construção da Matriz Triangular Inferior (Matriz):

Inicializa uma lista vazia (Matriz) para representar a matriz triangular inferior.
Utiliza loops para preencher cada linha da matriz com os coeficientes da lista A.

                    Resolução do Sistema Triangular:

Utiliza um loop para resolver o sistema linear usando a matriz triangular inferior.
Calcula o somatório dos produtos dos coeficientes da matriz pelos valores já encontrados de X.
Atualiza os valores de X usando a fórmula da matriz triangular inferior.

                    Formatação dos Resultados:

Formata os resultados para até 3 casas decimais na lista X.

                    Retorno dos Resultados (X):

Retorna a lista X com os valores calculados para as (n) incógnitas do sistema)

3) Função de exibição do sistema e dos resultados:
   Essa função deve organizar e exibir as listas A, B e X, geradas pelas duas últimas funções, de forma a apresentá-las como um sistema de (n) equações e (n) incógnitas. Além disso, deve exibir os resultados para cada incógnita encontrado pela função de resolução do sistema triangular. 
   
        Obtenção do Número de Equações:

Obtém o número de equações (n) a partir da lista B.

        Construção da Matriz Triangular Inferior:

Inicializa uma lista vazia (Matriz) que será preenchida com as linhas da matriz triangular inferior.
Utiliza loops para preencher cada linha da matriz com os coeficientes da lista A.

        Inicialização da String de Exibição:

Inicializa uma string vazia para armazenar a exibição do sistema de equações.

        Exibição do Sistema de Equações:

Utiliza loops para criar a representação do sistema de equações.
Adiciona cada termo formatado à linha do sistema.

        Exibição das Soluções (solucoes):

Inicializa uma string vazia para representar as soluções.
Utiliza loops para adicionar as soluções formatadas à string.
Adiciona espaços entre as soluções, exceto para a última.

        Composição da Exibição do Sistema:

Adiciona as linhas do sistema com termos independentes à exibição do sistema.
Adiciona as soluções à exibição do sistema.

            Exibição da Mensagem de Conclusão:
    
Imprime uma mensagem indicando que o sistema está pronto e resolvido.

            Retorno da Exibição do Sistema:

Retorna a string contendo a exibição do sistema e a resolução do mesmo.




ORGANIZAÇÃO DO PROGRAMA:

Função entrada:

Obtem e valida a entrada do usuário referente ao número de equações e os coeficientes/termos independentes.
Retorna as listas A (coeficientes) e B (termos independentes).
Função sistema_triangular(A, B):

Recebe as listas A e B e realiza o cálculo das incógnitas para um sistema triangular inferior.
Retorna a lista X contendo as soluções para as incógnitas.
Função exibe(A, B, X):

Recebe as listas A, B, e X, e exibe o sistema de equações, a matriz triangular inferior e as soluções.
Retorna uma string contendo a representação formatada do sistema resolvido.

Relação entre as Funções:

A função entrada obtém a entrada do usuário e retorna as listas A e B.
A função sistema_triangular recebe A e B, calcula as incógnitas e retorna X.
A função exibe recebe A, B, e X, e exibe o sistema resolvido.

Formato de Saída:

O programa exibe o sistema de equações e as soluções formatadas.
As soluções são exibidas com até 3 casas decimais.
A saída inclui uma mensagem indicando que o sistema está pronto e resolvido.

In [18]:
def entrada():
    """
    Solicita ao usuário a quantidade de equações do sistema,
    obtém os coeficientes e termos independentes e verifica
    a validade das entradas.

    Returns:
    A tuple (A, B) com as listas de coeficientes e termos independentes.
    """
    true = True # Uso de variável booleana para evitar repetições infinitas.
    while true: # Loop de verificação de entrada
        try:
            # Solicita ao usuário a quantidade de equações e verifica se é um inteiro positivo.
            entrada_usuario = input("Forneça a quantidade de equações do sistema: ") # Solicita ao usuário inserir a quantidade de equações do sistema e armazena a entrada na variável 

            if "." in entrada_usuario: #Verifica se há um ponto decimal na entrada do usuário.
                print("Por favor, insira um número inteiro positivo.") # Se o ponto decimal estiver presente, imprime uma mensagem solicitando um número inteiro positivo.
                continue  # Volta para o início do loop se a entrada contiver um ponto decimal (indicando um float).

            numero_de_equacoes = int(entrada_usuario) #transforma o valor para int, para que possa ser utilizado.

            if numero_de_equacoes <= 0: #número de equações não pode ser igual a zero ou negativo.
                print("O número de equações deve ser um inteiro maior que zero.") # Se o número de equações for inválido, imprime uma mensagem de erro.
            else: # Se o número de equações for válido (inteiro positivo), executa o bloco de código a seguir.
                true = False  #  Altera o valor de true para False, saindo do loop while.
        except ValueError: #Captura uma exceção do tipo ValueError que pode ocorrer se a conversão para inteiro falhar.
            print("Por favor, insira um número inteiro positivo.") # Imprime uma mensagem de erro para o caso em que a exceção ValueError ocorre.

    listacoeficientes = [] # Inicializa a lista que conterá os coeficientes do sistema.

    for i in range(1, numero_de_equacoes + 1):# Loop para obter os coeficientes de cada equação.
        linha_coef = [] #lista temporária que armazena os coeficientes de uma única linha da matriz.
        print(f"\nInsira os coeficientes da equação {i} (total de {i} coeficientes esperados):") #Solicita ao usuário inserir os coeficientes da equação atual e exibe uma mensagem informativa.
        for j in range(1, i + 1): # Inicia um loop para obter os coeficientes de cada coluna da equação atual. O loop percorre as colunas de 1 até i.
            while True: # Inicia um loop infinito para garantir que o usuário forneça um coeficiente válido.
                try: # Inicia um bloco de código que pode gerar exceções.
                    coeficiente = float(input(f"Coeficiente {j} da equação {i}: ")) # Solicita ao usuário o coeficiente da linha i, coluna j.
                except ValueError: #Captura a exceção ValueError que ocorre se a conversão falhar.
                    print("O coeficiente deve ser um número real. Tente novamente.") #Imprime uma mensagem de erro para o usuário.
                    continue # Volta para o início do loop while, pedindo novamente ao usuário que insira o coeficiente.
                else: # Se nenhum erro ocorreu durante o bloco try, executa o bloco de código a seguir.
                    break # Sai do loop while e continua para a próxima iteração do loop for.
            linha_coef.append(coeficiente) # Adiciona o coeficiente à lista temporária linha_coef.
        listacoeficientes.extend(linha_coef) # Estende a lista listacoeficientes com os coeficientes da linha atual.

    # O código verifica se a matriz é triangular inferior, e se não for, imprime uma mensagem de erro e reinicia o processo.
    for i in range(numero_de_equacoes):
        if listacoeficientes[int(i * (i + 1) / 2) + i] == 0:
            print("Os coeficientes de valor de linha e coluna iguais não podem ser zero. Reinicie o processo.")
            return entrada()

    listatermosindependentes = [] #Inicializa a lista que conterá os termos independentes.

    for i in range(numero_de_equacoes): #Inicia um loop para obter os termos independentes de cada equação.
        while True: # Inicia um loop infinito para garantir que o usuário forneça um termo independente válido.
            try:
                # Solicita ao usuário o termo independente da equação i.
                termo_independente = float(input(f"Termo independente da equação {i + 1}: ")) #Solicita ao usuário inserir um termo independente e converte a entrada para um número de ponto flutuante.
            except ValueError: # Captura a exceção ValueError que ocorre se a conversão falhar.
                print("O termo independente deve ser um número real. Tente novamente.") # Imprime uma mensagem de erro para o usuário.
                continue #Volta para o início do loop while, pedindo novamente ao usuário que insira o termo independente.
            else: #Se nenhum erro ocorreu durante o bloco try, executa o bloco de código a seguir.
                break #Sai do loop while e continua para a próxima iteração do loop for.
        listatermosindependentes.append(termo_independente) # Adiciona o termo independente à lista listatermosindependentes.

    print(f"\nCoeficientes: {listacoeficientes}\nTermos Independentes: {listatermosindependentes}\n") #Imprime as listas de coeficientes e termos independentes após a conclusão do processo.

    confirmação_para_passar = True #Inicializa uma variável chamada confirmação_para_passar com o valor True.
    while confirmação_para_passar: #Inicia um novo loop while que continuará até que confirmação_para_passar seja alterada para False.
        confirmar = int(input("Confirme os valores digitando 1. Para retornar, digite 0:")) #Solicita ao usuário que confirme os valores digitando 1 ou retorne digitando 0.
        if confirmar in {0, 1}: #Verifica se a entrada do usuário é 0 ou 1.
            confirmação_para_passar = False #Altera o valor de confirmação_para_passar para False, saindo do loop while.
    if confirmar == 1: #Se a confirmação do usuário for 1, executa o bloco de código a seguir.  
        A = listacoeficientes #Atribui a lista listacoeficientes à variável A.
        B = listatermosindependentes # Atribui a lista listatermosindependentes à variável B.

        return A, B # Retorna as listas A e B.
    else: # Se a confirmação do usuário for 0, executa:
        return entrada() # Chama recursivamente a função entrada() para reiniciar o processo.

def sistema_triangular(A, B): # Define a função sistema_triangular que recebe duas listas como parâmetros, A e B.
    """
    Resolve um sistema de equações lineares representado por uma matriz
    triangular inferior.

    Args:
    A (list): Lista de coeficientes do sistema (matriz triangular inferior).
    B (list): Lista de termos independentes.

    Returns:
    Uma lista X com as soluções do sistema.
    """
    n = len(B)  # Número de equações (e, portanto, de termos independentes)
    X = [0.0] * n #Inicializa a lista X que conterá as soluções, preenchida com zeros.

    # Construção da matriz diretamente
    Matriz = []  # Inicializa a lista que conterá a matriz triangular inferior
    for i in range(n): # Loop externo percorrendo cada linha da matriz.
        linha = [0.0] * n  # Inicializa cada linha com n zeros
        for j in range(i + 1): # Loop interno percorrendo cada coluna da matriz.
            linha[j] = A[int(i * (i + 1) / 2) + j] # Preenche a linha com os coeficientes da lista A seguindo a lógica da matriz triangular inferior.
        Matriz.append(linha) # Adiciona a linha preenchida à matriz.

    # Resolução do sistema triangular
    for i in range(n): # Loop externo para resolver o sistema linear usando a matriz triangular inferior.
        somatorio = 0 # Inicializa a variável somatorio que será usada para calcular a soma dos produtos dos coeficientes.
        for j in range(i): # Loop para calcular o somatório dos produtos dos coeficientes da matriz pelo vetor de soluções já encontradas.
            somatorio += Matriz[i][j] * X[j] #Calcula e acumula os produtos dos coeficientes da matriz pelo vetor de soluções.
        X[i] = (B[i] - somatorio) / Matriz[i][i] # Calcula o valor da incógnita atual usando o termo independente e o somatório, dividido pelo coeficiente associado à incógnita na matriz triangular.

    # Formatação dos resultados para até 3 casas decimais
    X = ["%.3f" % valor for valor in X]

    return X # Função retorna uma lista X com os valores calculados para as incógnitas.
(A, B) = entrada() #Nesse momento é feita a associação entre as funções, já que 
X = sistema_triangular(A,B)
def exibe(A, B, X): #Define a função exibe que recebe três listas como parâmetros, A, B e X.
    """
    Exibe o sistema de equações e as soluções.

    Args:
    A (list): Lista de coeficientes do sistema (matriz triangular inferior).
    B (list): Lista de termos independentes.
    X (list): Lista de soluções do sistema.

    Returns:
    Uma string com a representação do sistema e suas soluções.
    """
    n = len(B) #Obtém o número de equações (ou termos independentes) do sistema.

    Matriz = [] #Inicializa uma lista vazia que será preenchida com as linhas da matriz triangular inferior.
    for i in range(n): #Inicia um loop externo para percorrer cada linha da matriz triangular inferior.
        linha = [] # Inicializa uma lista vazia que representará uma linha da matriz.
        for j in range(n): # Inicia um loop interno para percorrer cada coluna da matriz.
            if j <= i: # Verifica se a coluna é menor ou igual à linha para preencher a matriz triangular inferior.
                linha.append(float(A[int(i * (i + 1) / 2) + j])) #Adiciona o coeficiente correspondente à posição (i, j) da matriz triangular inferior.
            else: #Caso contrário, preenche com 0.0.
                linha.append(0.0)
        Matriz.append(linha) # Adiciona a linha preenchida à matriz.

    exibicao_sistema = "" #Inicializa uma string vazia para armazenar a exibição do sistema de equações.

# Exibindo o sistema de equações
    for i in range(n): #Inicia um loop externo para exibir o sistema de equações.
        linha = "" # Inicializa uma string vazia para representar uma linha do sistema.
        for j in range(n): # Inicia um loop interno para percorrer cada coluna da matriz.
            if Matriz[i][j] != 0: #Verifica se o coeficiente é diferente de zero para incluir no sistema.
                if linha != "": # Adiciona um sinal de + se a linha não estiver vazia.
                    linha += " + "
                linha += f"({Matriz[i][j]:.3f})*X{j + 1}" # Cria o termo formatado para ser adicionado à linha.
            else:
                linha += " + 0.0" # Se o coeficiente for zero, adiciona " + 0.0".
        exibicao_sistema += linha + f" = {B[i]:.3f}\n" # Adiciona a linha do sistema com o termo independente à exibição do sistema.

    # Exibindo as soluções
    solucoes = "" # Inicializa uma string vazia para representar as soluções.
    for j in range(n): # Inicia um loop para exibir as soluções.
        solucoes += f"X{j + 1} = {float(X[j]):.3f}" #Adiciona a solução formatada à string.
        if j < n - 1: #Adiciona espaços entre as soluções, exceto para a última.
            solucoes += "  "

    exibicao_sistema += solucoes #Adiciona as soluções à exibição do sistema.

    print("\nO sistema está pronto e resolvido!\n") #Exibe uma mensagem indicando que o sistema está pronto e resolvido.

    return exibicao_sistema #Retorna a exibição do sistema.

print(exibe(A, B, X))





Forneça a quantidade de equações do sistema: 3

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 5

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: 8
Coeficiente 2 da equação 2: 9

Insira os coeficientes da equação 3 (total de 3 coeficientes esperados):
Coeficiente 1 da equação 3: -10
Coeficiente 2 da equação 3: 6
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: 4
Termo independente da equação 1: adf
O termo independente deve ser um número real. Tente novamente.
Termo independente da equação 1: 12
Termo independente da equação 2: 2.5
Termo independente da equação 3: -145

Coeficientes: [5.0, 8.0, 9.0, -10.0, 6.0, 4.0]
Termos Independentes: [12.0, 2.5, -145.0]

Confirme os valores digitando 1. Para retornar, digite 0:1

O sistema está 

UTILIZAÇÃO DO PROGRAMA:

        Iniciando o Programa:

Execute o programa em um ambiente Python.

        Entradas do Programa:

O programa solicitará a quantidade de equações no sistema.
Insira um número inteiro positivo quando solicitado.

        Entrada de Coeficientes:

Para cada equação, o programa pedirá os coeficientes.
Insira os coeficientes um por um, respondendo às solicitações.
Coeficientes devem ser números reais.  O programa entende apenas o uso de "." como decimal, pedindo para tentar novamente caso seja usado ",". 

        Entrada de Termos Independentes:

Para cada equação, o programa solicitará os termos independentes.
Insira os termos independentes um por um, respondendo às solicitações.
Termos independentes devem ser números reais. O programa entende apenas o uso de "." como decimal, pedindo para tentar novamente caso seja usado ",". 

            Confirmação dos Valores:

Após inserir todos os coeficientes e termos independentes, o programa exibirá os valores.
Confirme digitando 1 para prosseguir ou 0 para corrigir os valores.

            Resultado do Programa:

O programa calculará e exibirá o sistema resolvido, incluindo a matriz triangular inferior e as soluções.
As soluções serão apresentadas com até 3 casas decimais.

            Possíveis Erros e Mensagens:

Se inserir um número inválido de equações (não inteiro ou zero/negativo), o programa solicitará uma correção.
Se inserir coeficientes ou termos independentes não numéricos, o programa solicitará correção.
Se os coeficientes de valor de linha e coluna iguais forem zero, ou seja, a diagonal principal da matriz tiver algum valor nulo, o programa indicará um erro e reiniciará o processo, já que o sistema não apresenta resolução nesses casos.

TESTES FEITOS:
 1)   TESTES PARA ESCOLHA DE NÚMERO DE EQUAÇÕES 
 
 
    
             Cenário 1 (Entrada Válida):

Entrada do Usuário: 3 (Número válido de equações)
Resultado Esperado: O programa deve solicitar os coeficientes e termos independentes.


        RESPOSTA DO PROGRAMA

Forneça a quantidade de equações do sistema: 3

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1:



       Cenário 2 (Entrada Inválida - Não Inteira):

Entrada do Usuário: 2.5 (Número não inteiro)
Resultado Esperado: O programa deve solicitar uma correção, indicando que o número de equações deve ser um inteiro.

             RESPOSTA DO PROGRAMA 
             
Forneça a quantidade de equações do sistema: 2.5
Por favor, insira um número inteiro positivo.
Forneça a quantidade de equações do sistema:

        Cenário 3 (Entrada Inválida - Zero):

Entrada do Usuário: 0 (Número zero)
Resultado Esperado: O programa deve solicitar uma correção, indicando que o número de equações deve ser maior que zero.


        RESPOSTA DO PROGRAMA
Forneça a quantidade de equações do sistema: 0
O número de equações deve ser um inteiro maior que zero.
Forneça a quantidade de equações do sistema: 

         Cenário 4 (Entrada Inválida - Letras):
Entrada do Usuário: nais
Resultado Esperado: O programa deve solicitar uma correção, indicando é necessário entrar um número.

        RESPOSTA DO PROGRAMA
Forneça a quantidade de equações do sistema: nais
Por favor, insira um número inteiro positivo.
Forneça a quantidade de equações do sistema: 

2) TESTES PARA ESCOLHA DOS COEFICIENTES
        
        Cenário 1 (Entrada Válida - Número Real):
Entradas do Usuário: 2.6, -12, 9
Resultado Esperado: O programa deve aceitar os valores e solicitar os termos independentes

        RESPOSTA DO PROGRAMA 
Forneça a quantidade de equações do sistema: 2

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 2.6

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: -12
Coeficiente 2 da equação 2: 9
Termo independente da equação 1: 

        Cenário 2 (Entrada Inválida - Coeficiente com número igual ao da equação igual a 0)
   Nesse cenário o sistema não tem resolução, pois a matriz gerada tem termos nulos em sua diagonal principal
Entradas do usuário: 2.6, -12, 0
Resultado Esperado: O programa deve alertar sobre o erro e reiniciar as entradas

        RESPOSTA DO PROGRAMA
Forneça a quantidade de equações do sistema: 2

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 2.5

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: -12
Coeficiente 2 da equação 2: 0
Os coeficientes de valor de linha e coluna iguais não podem ser zero. Reinicie o processo.
Forneça a quantidade de equações do sistema:

            Cenário 3 (Entrada Inválida - Valor diferente de número real)
Entradas do usuário: 5, 8, 9, -10, 6, i
Resultado Esperado: O programa deve alertar sobre o erro e solicitar para recolocar o coeficiente específico com erro.

            RESPOSTA DO PROGRAMA
Forneça a quantidade de equações do sistema: 3

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 5

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: 8
Coeficiente 2 da equação 2: 9

Insira os coeficientes da equação 3 (total de 3 coeficientes esperados):
Coeficiente 1 da equação 3: -10
Coeficiente 2 da equação 3: 6
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3:

3) Testes para escolhas dos termos independentes 
Programa deve rejeitar qualquer valor que não seja um número real, e solicitar ao usuário para colocar o termo independente específico novamente. Quando o usuário colocar todos os termos independentes, é esperado que o programa mostre todos os coeficientes e termos independentes em ordem e solicite a confirmação para o calculo do sistema. 

        RESPOSTA DO PROGRAMA
        
 Forneça a quantidade de equações do sistema: 3

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 5

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: 8
Coeficiente 2 da equação 2: 9

Insira os coeficientes da equação 3 (total de 3 coeficientes esperados):
Coeficiente 1 da equação 3: -10
Coeficiente 2 da equação 3: 6
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: 4
Termo independente da equação 1: adf
O termo independente deve ser um número real. Tente novamente.
Termo independente da equação 1: 12
Termo independente da equação 2: 2.5
Termo independente da equação 3: -145

Coeficientes: [5.0, 8.0, 9.0, -10.0, 6.0, 4.0]
Termos Independentes: [12.0, 2.5, -145.0]

Confirme os valores digitando 1. Para retornar, digite 0:

4) Testes do funcionamento completo do programa, quando todas as entradas são válidas.

                Entradas:
Número de equações: 3
Coeficientes: [5.0, 8.0, 9.0, -10.0, 6.0, 4.0]
Termos Independentes: [12.0, 2.5, -145.0]

                Saída:
  (5.000)*X1 + 0.0 + 0.0 = 12.000
(8.000)*X1 + (9.000)*X2 + 0.0 = 2.500
(-10.000)*X1 + (6.000)*X2 + (4.000)*X3 = -145.000
X1 = 2.400  X2 = -1.856  X3 = -27.467
                
                RESPOSTAS DO PROGRAMA
  
 Forneça a quantidade de equações do sistema: 3

Insira os coeficientes da equação 1 (total de 1 coeficientes esperados):
Coeficiente 1 da equação 1: 5

Insira os coeficientes da equação 2 (total de 2 coeficientes esperados):
Coeficiente 1 da equação 2: 8
Coeficiente 2 da equação 2: 9

Insira os coeficientes da equação 3 (total de 3 coeficientes esperados):
Coeficiente 1 da equação 3: -10
Coeficiente 2 da equação 3: 6
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: i
O coeficiente deve ser um número real. Tente novamente.
Coeficiente 3 da equação 3: 4
Termo independente da equação 1: adf
O termo independente deve ser um número real. Tente novamente.
Termo independente da equação 1: 12
Termo independente da equação 2: 2.5
Termo independente da equação 3: -145

Coeficientes: [5.0, 8.0, 9.0, -10.0, 6.0, 4.0]
Termos Independentes: [12.0, 2.5, -145.0]

Confirme os valores digitando 1. Para retornar, digite 0:1

O sistema está pronto e resolvido!

(5.000)*X1 + 0.0 + 0.0 = 12.000
(8.000)*X1 + (9.000)*X2 + 0.0 = 2.500
(-10.000)*X1 + (6.000)*X2 + (4.000)*X3 = -145.000
X1 = 2.400  X2 = -1.856  X3 = -27.467
                
                


CONCLUSÃO:

Este trabalho prático se revelou como uma jornada enriquecedora no universo da programação em Python. Ao longo do processo, a necessidade de aprimorar minhas habilidades me impulsionou a estudar e pesquisar para encontrar soluções diferentes eficientes. Essa busca pelo conhecimento não apenas consolidou conceitos já familiares, mas também me introduziu a novas técnicas e ferramentas.

A interação com o usuário, ponto central do programa, demandou uma compreensão da manipulação de entrada e saída, incentivando-me a explorar estruturas de repetição e condições de maneira mais refinada. A utilização de matrizes e laços 'for' aninhados para percorrê-las ofereceu um desafio, evidenciando a versatilidade da linguagem Python. Além disso, a necessidade de encontrar e prever as possíveis falhas que o programa pode apresentar para um usuário me obrigaram a entender exatamente o que o programa estava fazendo, sendo este o código em que eu tive mais consciencia sobre o que estava fazendo e como o programa entende e lida com as informações que está recebendo. 

Uma ênfase significativa foi dada à formatação, visando não apenas à correta execução do código, mas também à apresentação clara e organizada das informações para o usuário. A formatação foi crucial em diversos aspectos, desde a entrada de dados até a exibição do sistema de equações e suas soluções.

No processo de entrada, foram aplicadas verificações para garantir que o usuário inserisse valores válidos, evitando possíveis erros e mantendo a execução do programa suave. A utilização de mensagens formatadas e estruturas de repetição contribuiu para uma interação mais amigável, orientando o usuário de forma clara e concisa.

A conexão entre as funções, da entrada à exibição do sistema, destacou a importância de uma visão coesa e abrangente do código. A prática da documentação, especialmente através de docstrings, emergiu como uma ferramenta fundamental para a clareza e compreensão do código.